standard.md

```diff
@@ -0,0 +1,114 @@
+
+### Reserved options
+
+#### Related to the dataset
+
+- **project**: `str`. Name of the project at which this dataset belongs. It can be a real project, like `CMIP5`, `CMIP6` or a tool defined agrupation of datasets like `Observations` or `Reanalysis`
+- **activity**: `str`. CMIP activity or analog for the current data
+- **institute**: `list(str)`. List of institutes associated to the current data.
+- **dataset**: `str`. Name of the dataset, being it the CMIP model indetifier, the name of an observation
+- **ensemble**: `str`. Ensemble identifier for the current data
+
+#### Related to the variable
+
+- **table**: `str`. Original mip of the current variable
+- **frequency**: `str`. Original frequency of the data used
+- **modeling_realm**: `list(str)`. List of realms at which the variable belonh
+- **grid**: `str`. Original grid of the variable
+- **units**: `str`. Units of the data
+- **short_name**: `str`. Variable name used in the file.
+- **standard_name**: `str`. Standard name of the variable.
+- **long_name**: `str`. Long name of the variable
+
+#### Others
+
+- **start**: `str`. Starting date of the data, in the format YYYYMMDD
+- **end**: `str`, Ending date of the data, in the format YYYYMMDD
+- **start_year**: `int`. Starting year of the data
+- **end_year**: `int`. End year of the data
+- **reference_dataset**: `str`. Dataset to be used as the reference to compare the current dataset with, given as the alias.
+
+## The script formal description file
+
+It is recommended that each script comes with a description file in yaml syntax which includes :
+- **script_name**: `str`. A label for the name of the script, which uniquely identifies it, and is delivered by an authoritative entity (such as an ESGF, ENES or WCRP entitiy)
+- **script_interface_version**: `str`. A label for distinguishing among various version of the script interface
+-  **mandatory_keys**: `list(str)`. the list of reserved and custom keys for which the tool must provide values
+- **input_type**: `str`. which can take values 'member', 'ensemble' and 'any', and which allows the tool to check that values provided to the script are of the right type. 'ensemble' means : an ensemble with more than one member, while 'any' allows for an ensemble of size 1. If the type differs among the variables, this entry can be a dictionary with key = variable name
+- **outputs**:`dict`. A dictionary of patterns  which allows the tool to discover every output file and to assign a label to it. The patterns and the labels used as dictionary keys  can make use of keywords 'variable', 'dataset' and 'reference_dataset'; the tool may instantiate the keywords with all possible values to form file basenames and test whether a corresponding file exists in scripts output directories; dictionary values can be pairs, in which case, the second element of the pair allows for providing a pattern for indicating the output variable's short_name
```

Comment:
I'm not sure if this is the right approach, diagnostic developers may struggle to give their output files the names they would like and may be unable to provide useful metadata about the output files. I would rather go for a standard output metadata format, e.g. the same or very similar to the input metadata format. I would recommend adding things like caption, authors, references, input files used, etc like we do in the ESMValCore: https://docs.esmvaltool.org/projects/esmvalcore/en/latest/interfaces.html#information-provided-by-the-diagnostic-script-to-esmvalcore

Reply:
If the tool has no way to predict the list of possible script output files (and to name them) without running the script, it cannot implement a cache of script results (because he must run the script anyway)
This is a basic shortcoming of ESMValTool

Reply:
> diagnostic developers may struggle to give their output files the names they would like
This appears self-contradictory

Reply:
See #11 for continuing the discussion about how to describe script's outputs